src/credentials/FabricTable.cpp

```diff
@@ -41,145 +42,156 @@ CHIP_ERROR FabricInfo::SetFabricLabel(const CharSpan & fabricLabel)
     {
+        Crypto::P256SerializedKeypair serializedOpKey;
+        uint16_t size = serializedOpKey.Capacity();
+        ReturnErrorOnFailure(storage->SyncGetKeyValue(keyAlloc.FabricOpKey(mFabric), serializedOpKey.Bytes(), size));
+        serializedOpKey.SetLength(size);
+        if (mOperationalKey == nullptr)
+        {
 #ifdef ENABLE_HSM_CASE_OPS_KEY
-        mOperationalKey = chip::Platform::New<P256KeypairHSM>();
-        mOperationalKey->SetKeyId(CASE_OPS_KEY);
+            mOperationalKey = chip::Platform::New<P256KeypairHSM>();
+            mOperationalKey->SetKeyId(CASE_OPS_KEY);
 #else
-        mOperationalKey = chip::Platform::New<P256Keypair>();
+            mOperationalKey = chip::Platform::New<P256Keypair>();
 #endif
-    }
-    VerifyOrExit(mOperationalKey != nullptr, err = CHIP_ERROR_NO_MEMORY);
-    SuccessOrExit(err = mOperationalKey->Deserialize(info->mOperationalKey));
+        }
+        VerifyOrReturnError(mOperationalKey != nullptr, CHIP_ERROR_NO_MEMORY);
+        ReturnErrorOnFailure(mOperationalKey->Deserialize(serializedOpKey));
 #ifdef ENABLE_HSM_CASE_OPS_KEY
-    // Set provisioned_key = true , so that key is not deleted from HSM.
-    mOperationalKey->provisioned_key = true;
+        // Set provisioned_key = true , so that key is not deleted from HSM.
+        // mOperationalKey->provisioned_key = true;
 #endif
+    }
 
-    ChipLogProgress(Inet, "Loading certs from storage");
-    SuccessOrExit(err = SetRootCert(ByteSpan(info->mRootCert, rootCertLen)));
+    {
+        uint8_t buf[MetadataTLVMaxSize()];
+        uint16_t size = sizeof(buf);
+        ReturnErrorOnFailure(storage->SyncGetKeyValue(keyAlloc.FabricMetadata(mFabric), buf, size));
+        TLV::TLVReader reader;
+        reader.Init(buf, size);
 
-    // The compressed fabric ID doesn't change for a fabric over time.
-    // Computing it here will save computational overhead when it's accessed by other
-    // parts of the code.
-    SuccessOrExit(err = ExtractNodeIdFabricIdFromOpCert(ByteSpan(info->mNOCCert, nocCertLen), &nodeId, &mFabricId));
-    SuccessOrExit(err = GeneratePeerId(mFabricId, nodeId, &mOperationalId));
+        ReturnErrorOnFailure(reader.Next(TLV::kTLVType_Structure, TLV::AnonymousTag()));
+        TLV::TLVType containerType;
+        ReturnErrorOnFailure(reader.EnterContainer(containerType));
 
-    SuccessOrExit(err = SetICACert(ByteSpan(info->mICACert, icaCertLen)));
-    SuccessOrExit(err = SetNOCCert(ByteSpan(info->mNOCCert, nocCertLen)));
+        ReturnErrorOnFailure(reader.Next(kVendorIdTag));
+        ReturnErrorOnFailure(reader.Get(mVendorId));
 
-exit:
-    if (info != nullptr)
-    {
-        chip::Platform::Delete(info);
+        ReturnErrorOnFailure(reader.Next(kFabricLabelTag));
+        CharSpan label;
+        ReturnErrorOnFailure(reader.Get(label));
+
+        VerifyOrReturnError(label.size() <= kFabricLabelMaxLengthInBytes, CHIP_ERROR_BUFFER_TOO_SMALL);
+        Platform::CopyString(mFabricLabel, label);
```

Comment:
Why is CopyString using `Platform::` ? Seems wrongly namespaced.

Reply:
Plausible, but way out of scope for this PR.